src/apps/bridge/reportBuilder.cpp

```diff
@@ -0,0 +1,497 @@
+ * @param sample_counter The current sample counter, indicating the position at which to add the sensor data.
+ */
+void ReportBuilderLinkedList::addSensorDataToElement(report_builder_element_t *element, uint8_t sensor_type, void *sensor_data, uint32_t sample_counter) {
+    switch (sensor_type) {
+      case AANDERAA_SENSOR_TYPE: {
+        // Back fill the sensor_data with NANs if we are not on the right sample count
+        uint32_t i;
+        for (i = 0; i < sample_counter; i++) {
+          memcpy(&(static_cast<aanderaa_aggregations_t *>(element->sensor_data))[i], &NAN_AGG, sizeof(aanderaa_aggregations_t));
+        }
+        // Copy the sensor data into the elements array in the correct location within the buffer
+        // If it is NULL then just fill it with NAN again
+        if (sensor_data != NULL) {
+          memcpy(&(static_cast<aanderaa_aggregations_t *>(element->sensor_data))[i], sensor_data, sizeof(aanderaa_aggregations_t));
+        } else {
+          memcpy(&(static_cast<aanderaa_aggregations_t *>(element->sensor_data))[i], &NAN_AGG, sizeof(aanderaa_aggregations_t));
+        }
+        break;
+      }
+      default: {
+        printf("Unknown sensor type in addSensorDataToElement\n");
+        configASSERT(0);
+        break;
+      }
+    }
+}
+
+/**
+ * @brief Adds a sample to a specific element in the report builder linked list.
+ *
+ * This function finds the element associated with the given node_id in the linked list. If the element is found, it adds the sensor data to the element. If the element is not found, it creates a new element and appends it to the linked list.
+ *
+ * @param node_id The ID of the node for the report.
+ * @param sensor_type The type of the sensor for the report.
+ * @param sensor_data Pointer to the sensor data for the report.
+ * @param sensor_data_size The size of the sensor data.
+ * @param samples_per_report The number of samples per report.
+ * @param sample_counter The current sample counter.
+ * @return A boolean value indicating whether the operation was successful. Returns true if the sample was added successfully, false otherwise.
+ */
+bool ReportBuilderLinkedList::findElementAndAddSampleToElement(uint64_t node_id, uint8_t sensor_type, void *sensor_data, uint32_t sensor_data_size, uint32_t samples_per_report, uint32_t sample_counter) {
+  bool rval = false;
+  report_builder_element_t *element = findElement(node_id);
+  if (element != NULL) {
+    addSensorDataToElement(element, sensor_type, sensor_data, sample_counter);
+    rval = true;
+  } else {
+    if (head == NULL && size == 0) {
+      append(NULL, node_id, sensor_type, sensor_data, sensor_data_size, samples_per_report, sample_counter);
+      rval = true;
+    } else {
+      report_builder_element_t *element = head;
+      while (element->next != NULL) {
+        element = element->next;
+      }
+      append(element, node_id, sensor_type, sensor_data, sensor_data_size, samples_per_report, sample_counter);
```

Comment:
do we need an `rval = true` after this append? is there ever a scenario where this is false?

Reply:
Nope, also the functions that `findElementAndAddSampleToElement` call don't return anything so I'll also make this function a void return type.